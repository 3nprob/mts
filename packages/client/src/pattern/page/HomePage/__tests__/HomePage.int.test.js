import React from 'react'
import { async, regular } from './HomePage.int.story'

jest.mock('react-router-dom', () => ({ Link: () => <div /> }))
jest.mock('../../../../../../library/src/client/navBar', () => ({ handleStickyNavBar: () => {} }))

describe('<HomePage />', () => {
  afterEach(() => jest.clearAllMocks())

  describe('Render', () => {
    test('must match async', () => {
      expect(global.renderToJSON(async())).toMatchSnapshot()
    })

    test('must match regular', () => {
      expect(global.renderToJSON(regular())).toMatchSnapshot()
    })
  })
})
