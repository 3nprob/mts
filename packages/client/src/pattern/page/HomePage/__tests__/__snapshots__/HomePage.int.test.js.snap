// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> Snaphot must match asyncRender 1`] = `null`;

exports[`<HomePage /> Snaphot must match defaultRender 1`] = `
<div
  className=""
>
  <div
    className="homeTemplate"
  >
    <nav
      className="nav"
    >
      <div
        className="navBar"
      >
        <a
          className="navBarLogo"
          href="/"
        >
          <img
            alt="logo"
            height="40"
            src="/"
          />
        </a>
        <ul
          className="navBarList"
        >
          <li
            className="navBarListItem"
          >
            <a
              className="navBarListItemLink"
              href="/about"
            >
              About
            </a>
          </li>
          <li
            className="navBarListItem"
          >
            <a
              className="navBarListItemLink"
              href="/blog"
            >
              Blog
            </a>
          </li>
          <li
            className="navBarListItem"
          >
            <a
              className="navBarListItemLink"
              href="/courses"
            >
              Courses
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <main
      className="main"
    >
      <div
        className="mainModule"
      >
        <div
          className="mainBackground"
        >
          <picture>
            <img
              alt="cover"
              className="mainBackgroundImage"
              src={Object {}}
            />
          </picture>
          <div
            className="mainBackgroundDimmer"
          />
          <div
            className="mainBackgroundCaption"
          >
            <h1
              className="mainBackgroundCaptionPrimaryHeader"
            >
              MTS
            </h1>
            <h2
              className="mainBackgroundCaptionSecondaryHeader"
            >
              React.js, Node.js and GraphQL
            </h2>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
